train.py

```diff
@@ -51,16 +51,18 @@ def setup(rank, world_size, port):
     dist.init_process_group("nccl", rank=rank, world_size=world_size)
 
 
-def optim(opt, params, lr, betas):
+def optim(opt, params, lr, betas, weight_decay):
```

Comment:
weight_decay is 5th param: https://pytorch.org/docs/stable/generated/torch.optim.Adam.html